src/components/Views/SyncView/SyncView.js

```diff
@@ -84,10 +85,14 @@ class SyncView extends React.Component<
   }
 
   componentWillUnmount() {
+    const { unannounceSync } = nextProps;
+
     if (this.interval) {
       clearInterval(this.interval);
     }
 
+    unannounceSync();
```

Comment:
Is it guaranteed that `componentWillUnmount` will be called when the view closes?

Reply:
No, it is not guaranteed that the view will unmount. You might need to hook into the willBlur listener on react-navigation though: https://reactnavigation.org/docs/en/navigation-prop.html#addlistener-subscribe-to-updates-to-navigation-lifecycle

Reply:
@czengg If we are then announcing on `componentDidMount` but unannouncing on `willBlur`, could we get into a state where we can "unblur" without re-mounting and have our announcing be turned off?

I'm feeling a little out of my depth here, and karissa is taking some time off. Looking to follow your lead on this. :)

Reply:
@noffle Sorry just saw this. I can take over this branch if you'd like. We will need to check willFocus as well to trigger an announce. We shouldn't rely on the component lifecycle for this